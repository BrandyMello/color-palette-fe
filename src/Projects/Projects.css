td {
  border: 1px solid black;
  border-radius: 8px;
  width: 300px;
  height:300px;
}

table {
  margin: 2%;
}

.project-name-h2 {
  font-size: 40px;
  text-align: center;
}

.palette-name {
  font-size: 30px;
}

.delete-palette {
  background-color: #808080;
  color: white;
  border: 1px solid black;
  border-radius: 5px;
  font-size: 14px;
  height: 40px;
  width: 100px;
}