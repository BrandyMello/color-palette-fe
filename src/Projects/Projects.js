import React from 'react';
import './Projects.css';

const Projects = (props) => {
  console.log('props', props)
  let projectPalettes = props.palettes.filter(palette => {
    if(palette.projectName === props.name) {
      return palette
    }
  });
  let paletteRow = projectPalettes.map((projPalette, index) => {
   return (
     <>
     <tr>
       <th>{projPalette.name}</th>
    {/* <button type="submit" onclick={props.deleteSpecificProject(props.id)}>Delete Project</button> */}
     </tr>
   <tr key={projPalette.id}>
     <td key={index} style={{ backgroundColor: projPalette.colorOne }}>lock</td>
     <td key={index} style={{ backgroundColor: projPalette.colorTwo }}>lock</td>
     <td key={index} style={{ backgroundColor: projPalette.colorThree }}>lock</td>
     <td key={index} style={{ backgroundColor: projPalette.colorFour }}>lock</td>
     <td key={index} style={{ backgroundColor: projPalette.colorFive }}>lock</td>
     <button type="submit" onClick={() => props.deleteSpecificPalette(props.palettes[index].id)}>Delete Palette</button>
    </tr>
    </>
   )
  })
  return (
    <div>
      <h2 contentEditable={true}>{props.name}</h2>
      <table>
        {paletteRow}
      </table>
    </div>
  )
}

export default Projects