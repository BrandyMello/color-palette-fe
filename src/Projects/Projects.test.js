import React from 'react';
import { shallow } from 'enzyme';
import Projects from './Projects';

describe('Projects', () => {
  it('should match the snapshot', () => {
    const mockProjects = [{
      id: 1,
      name: 'Warm Colors'
    }, {
      id: 2,
      name: 'Cool Colors'
    }]

    const mockProps = { 
      id: 3, 
      name: "Neutral Colors", 
      created_at: "2019-10-08T23:00:51.517Z", updated_at: "2019-10-08T23:00:51.517Z", palettes: [{
        colorFive: "#472C25",
        colorFour: "#91371B",
        colorOne: "#F7E5D4",
        colorThree: "#EB712F",
        colorTwo: "#FADDAF",
        name: "Spiced",
        projectId: 1
      }, {
          colorFive: "#472C25",
          colorFour: "#91371B",
          colorOne: "#F7E5D4",
          colorThree: "#EB712F",
          colorTwo: "#FADDAF",
          name: "Spiced",
        projectId: 1
        }] }
    const wrapper = shallow(<Projects
      {...mockProps}  
      palettes={mockProjects}
      updateProjectName={jest.fn()}
      deleteSpecificProject={jest.fn()}
      deletespecificPalette={jest.fn()}
    />);

    expect(wrapper).toMatchSnapshot();
  });
})