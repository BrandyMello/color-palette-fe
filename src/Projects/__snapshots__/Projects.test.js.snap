// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects should match the snapshot 1`] = `
<div>
  <h2
    className="project-name-h2"
    contentEditable={true}
    onKeyUpCapture={[Function]}
    suppressContentEditableWarning={true}
  >
    Neutral Colors
  </h2>
  <button
    className="delete-project"
    onClick={[Function]}
    type="submit"
  >
    Delete Project
  </button>
  <table />
</div>
`;
